source/server/init_manager_impl.h

```diff
@@ -29,6 +29,7 @@ class InitManagerImpl : public Init::Manager, Logger::Loggable<Logger::Id::init>
 
   void initializeTarget(TargetWithDescription& target);
 
+  std::shared_ptr<InitManagerImpl*> self_; // For safe use by callbacks passed to init targets.
```

Comment:
This is the crux of this change. The init manager and all callbacks it creates have a shared_ptr to a pointer, which will either be `this` or `nullptr`.